mlrun/datastore/targets.py

```diff
@@ -70,6 +71,49 @@ def get_default_prefix_for_target(kind):
     return data_prefix
 
 
+def validate_target_list(targets):
+    """Check that no target overrides another target in the list (name/path)"""
+
+    if not targets:
+        return
+    targets_by_kind_name = [kind for kind in targets if type(kind) is str]
+    no_name_target_types_count = Counter(
+        [
+            target.kind
+            for target in targets
+            if hasattr(target, "name") and hasattr(target, "kind") and not target.name
+        ]
+        + targets_by_kind_name
+    )
+    overriding_name_target_types = [
+        t[0] for t in no_name_target_types_count.items() if t[1] > 1
+    ]
+    if overriding_name_target_types:
+        raise mlrun.errors.MLRunInvalidArgumentError(
+            "Only one default name per target type is allowed (please specify name for {0} target)".format(
+                overriding_name_target_types
+            )
+        )
+
+    no_path_target_types_count = Counter(
+        [
+            target.kind
+            for target in targets
+            if hasattr(target, "path") and hasattr(target, "kind") and not target.path
+        ]
+        + targets_by_kind_name
+    )
+    overriding_path_target_types = [
+        t[0] for t in no_path_target_types_count.items() if t[1] > 1
+    ]
+    if overriding_path_target_types:
+        raise mlrun.errors.MLRunInvalidArgumentError(
+            "Only one default path per target type is allowed (please specify path for {0} target)".format(
+                overriding_path_target_types
+            )
+        )
+
```

Comment:
You're only verifying here the targets that gets the default name
What about verifying that the case where you received several targets, that has names, but more than one target has the same name ?